tests/v2/validation/registries/registry_test.go

```diff
@@ -0,0 +1,221 @@
+package registries
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/rancher/rancher/tests/framework/clients/corral"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+
+	corralext "github.com/rancher/rancher/tests/framework/extensions/corral"
+	ranchersetup "github.com/rancher/rancher/tests/framework/extensions/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/registries"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	corralRegistryRancherImageName = "registryrancher"
+	corralRegistryRancherName      = "rancherglobalregistry"
+	corralAuthDisabledImageName    = "registryauthdisabled"
+	corralAuthDisabledName         = "downstreamregistrydisabled"
+	corralAuthEnabledImageName     = "registryauthenabled"
+	corralAuthEnabledName          = "downstreamregistryenabled"
+)
+
+var (
+	adminPassword = os.Getenv("ADMIN_PASSWORD")
+)
+
+type RegistryTestSuite struct {
+	suite.Suite
+	session                        *session.Session
+	podListClusterAuth             *v1.SteveCollection
+	podListClusterNoAuth           *v1.SteveCollection
+	podListClusterLocal            *v1.SteveCollection
+	clusterAuthRegistryHost        string
+	clusterNoAuthRegistryHost      string
+	localClusterGlobalRegistryHost string
+}
+
+func (rt *RegistryTestSuite) TearDownSuite() {
+	ranchersetup.RancherCleanup(rt.session)
+	corralList, err := corral.ListCorral()
+	require.NoError(rt.T(), err)
+	err = corralext.CleanUpCorrals(corralList)
+	require.NoError(rt.T(), err)
+	rt.session.Cleanup()
+}
+
+func (rt *RegistryTestSuite) SetupSuite() {
```

Comment:
So I've got a question about this setup suite, whenever we want to run this suite or test cases, this setup suite going to create private registries and HA again. What do you all think, personally, I think these resource creations shouldn't be happening all the time. Cause I can see where we spin up the resources but want to re-run only the tests?

Reply:
I totally agree with you. I think the rancher HA and post setup steps should be done outside of the SetupSuite, in possibly a main.go file that multiple test suites can use, like you had before I was just meant it needed to be in a shared directory. You should have two main.go files rancherha and ranchercleanup. A lot of tests will need to do that, and doing it here will limit the registry test to always create a rancher setup. Does that make sense?